### PR TITLE
fix null reference and shows entire frame bug.

### PR DESCRIPTION
**Pull request by ghost**
This fix bug wrote in Title.

### CODE REVIEW

File: js/rpg_core/Sprite.js [hunk 1]
```diff
@@ -287,7 +298,7 @@ Sprite.prototype._refresh = function() {
  */
 Sprite.prototype._isInBitmapRect = function(x, y, w, h) {
     return (this._bitmap && x + w > 0 && y + h > 0 &&
-            x < this._bitmap.width && y < this._bitmap.height);
+        x < this._bitmap.width && y < this._bitmap.height);
```
Comment (krmbn0576) [inline js/rpg_core/Sprite.js]: Found format-only changes! :-)

Comment (ghost) [reply js/rpg_core/Sprite.js]: OK fixed it.

File: js/rpg_core/Sprite.js [hunk 2]
```diff
@@ -222,10 +227,16 @@ Sprite.prototype.setColorTone = function(tone) {
  * @private
  */
 Sprite.prototype._onBitmapLoad = function() {
-    if (this._frame.width === 0 && this._frame.height === 0) {
-        this._frame.width = this._bitmap.width;
-        this._frame.height = this._bitmap.height;
+    if(this._requestBitmap === this._bitmap){
```
Comment (krmbn0576) [inline js/rpg_core/Sprite.js]: I don't know why `this._requestBitmap` is defined, because it's always the same as `this._bitmap`.

**Comment by krmbn0576** [comment]
I found a description similar to `TillingSprite#bitmap(set)`. Would you like to change it the same way?

File: js/rpg_core/Sprite.js (hunk 2 above)
Comment (ghost) [reply js/rpg_core/Sprite.js]: nope. _onBitmapLoad is async.
There is some chance to change Bitmap while loading is not completed.

**Comment by ghost** [comment]
TilingSprite is already guarded by line: 183
I think this is enough.

File: js/rpg_core/Sprite.js (hunk 2 above)
Comment (krmbn0576) [reply js/rpg_core/Sprite.js]: Of course it's async, so bitmap may be changed.
But... `this._requestBitmap` is changed at the same time. (69-70)
So, this condition-expression will always be `true`.

**Comment by krmbn0576** [comment]
OH, really?
If so, why not `Sprite` as well??
It's one-line!

File: js/rpg_core/Sprite.js (hunk 2 above)
Comment (ghost) [reply js/rpg_core/Sprite.js]: Oops you are right.
Something went wrong, I'll re-test it.

**Comment by ghost** [comment]
Sprite needs clearing frame, This can cause showing entire frame.
so I defer clearing frame until Bitmap will load.

**Comment by krmbn0576** [comment]
Hmm, I still don't understand your intentions.
What situations do you want to avoid?

In this code, frame is refreshed when bitmap changes null to non-null.
frame is never changed when bitmap changes non-null to anything.
Is this behavior correct?

**Comment by ghost** [comment]
I deferred only clearing entire frame.
updating frame is done by `_refresh`

File: js/rpg_core/Sprite.js [hunk 3]
```diff
@@ -57,13 +57,17 @@ Object.defineProperty(Sprite.prototype, 'bitmap', {
...
     set: function(value) {
         if (this._bitmap !== value) {
-            this._bitmap = value;
-            if (this._bitmap) {
-                this.setFrame(0, 0, 0, 0);
-                this._bitmap.addLoadListener(this._onBitmapLoad.bind(this));
-            } else {
+            if(!this._bitmap){
+                this._refreshFrame = true;
+            }else if(this._bitmap && value){
```
Comment (krmbn0576) [inline js/rpg_core/Sprite.js:62]: this condition-expression can be written shorter.
Written long for readability?

File: js/rpg_core/Sprite.js [hunk 4]
```diff
@@ -221,11 +225,15 @@ Sprite.prototype.setColorTone = function(tone) {
  * @method _onBitmapLoad
  * @private
  */
-Sprite.prototype._onBitmapLoad = function() {
-    if (this._frame.width === 0 && this._frame.height === 0) {
-        this._frame.width = this._bitmap.width;
-        this._frame.height = this._bitmap.height;
+Sprite.prototype._onBitmapLoad = function(bitmapLoaded) {
+    if(bitmapLoaded === this._bitmap){
+        if (this._refreshFrame && this._bitmap) {
```
Comment (krmbn0576) [inline js/rpg_core/Sprite.js:230]: I think `this._bitmap` is always non-null because it equals `bitmapLoaded`.
So, it's enough `if (this._refreshFrame)`

Comment (ghost) [reply js/rpg_core/Sprite.js:230]: Nope onBitmapLoad is async. there are some chance `bitmapLoaded !== this._bitmap`

File: js/rpg_core/Sprite.js (hunk 3 above)
Comment (ghost) [reply js/rpg_core/Sprite.js:62]: Yes I prefer self-descriptive code.

File: js/rpg_core/Sprite.js (hunk 4 above)
Comment (krmbn0576) [reply js/rpg_core/Sprite.js:230]: Oh sorry, I talked about line 230 only.
I thought ` && this._bitmap` isn't necessary because it's always non-null.

Comment (ghost) [reply js/rpg_core/Sprite.js:230]: If  We approve your suggestion, The reader must know about behavior of bitmapLoaded.

**Review by krmbn0576 — APPROVED** [review]
OK, LGTM! 🍣

**Pull request merged**